gossip/tx_scrambler_test.go

```diff
@@ -0,0 +1,597 @@
+package gossip
+
+import (
+	"fmt"
+	"github.com/ethereum/go-ethereum/common"
+	"golang.org/x/exp/rand"
+	"math"
+	"reflect"
+	"testing"
+)
+
+func TestTxScrambler_AnalyseEntryList_RemovesDuplicateTransactions(t *testing.T) {
+	entries := []*scramblerEntry{
+		{hash: common.Hash{1}},
+		{hash: common.Hash{2}},
+		{hash: common.Hash{3}},
+		{hash: common.Hash{2}},
+		{hash: common.Hash{1}},
+	}
+
+	shuffleEntries(entries)
+	result, _, _ := analyseEntryList(entries)
+	if len(result) == 0 {
+		t.Fatal("analyseEntryList returned empty list")
+	}
```

Comment:
This can be stricter:
```suggestion
	if len(result) != 3 {
		t.Fatalf("unexpected length of result list, wanted 3, got %d", len(result))
	}
```